### ISSUE TITLE
Channel state is CHANNELD_NORMAL but channels are not visible

### ISSUE THREAD

**Issue by kubilay902** [ElementsProject/lightning#837]
My node id is;
`02626c42851a22ba4c81c8e2f3c8dffd4634df777183c717da5d0597e238971e1f`

I created a channel few days ago at first everything seemed ok to me. But after a bit investigation and cross checks I saw that my channels are not ok as they seems in the listpeers command. listchannel command is not even listing those channels.

This is listpeers output of one of my peers; txid is valid but channels are gone. I try to refund the channel but it says it already exists.
```
{
            "channels": [
                {
                    "channel_reserve_satoshis": 0,
                    "dust_limit_satoshis": 546,
                    "funding_txid": "7669b0853a7be1563ba0b77c877c04a456795e2c26238a0a2ab6681988f2d1b6",
                    "htlc_minimum_msat": 0,
                    "max_accepted_htlcs": 483,
                    "max_htlc_value_in_flight_msat": 18446744073709551615,
                    "msatoshi_to_us": 80000000,
                    "msatoshi_total": 80000000,
                    "owner": "lightning_channeld",
                    "short_channel_id": "506368:1209:1",
                    "state": "CHANNELD_NORMAL",
                    "to_self_delay": 144
                }
            ],
            "connected": true,
            "id": "0371190acfb2e92bd1faa6ce4d12ff248798515a92ac903ac14d31b5172d9b2917",
            "netaddr": [
                "176.9.113.240:9735"
            ]
        }
```

I hope this is not a conflict issue. If you point me out the some of the functions to look at I can try to solve or investigate the issue too.

**Comment by cdecker** [comment]
There is a difference between the `listpeers` and the `listchannels` commands:

 - `listpeers` will show you your direct peers with whom you have a connection or a channel
 - `listchannels` will show all known channels in the network, that may or may not include your own channels

Your own channels may be missing from the `listchannels` output for a number of reasons, if for example the channel hasn't reached the announcement depth of 6 blocks, your channel will not be announced publicly. With around 260 confirmations I guess that's not the case for you. Another may be that the counterparty did not opt in to make the channel public, or that the announcement got lost somehow.

**Comment by kubilay902** [comment]
Hmm, if we wrap up; `listpeers` shows current channels even if it is not public. `listchannels` lists all public channels. Then we need an api like `mychannels` to see all public and non public channels. Is there any way to track historical channel data?
Thanks for response. I will dig into logs to see if there is any chance to find out what happened.

**Comment by cdecker** [comment]
`listpeers <id> debug` will also include logs (since the last restart) corresponding to the peer with nodeid `<id>`.

**Comment by kubilay902** [comment]
Hmm here is the Logs;

`{
    "peers": [
        {
            "channels": [
                {
                    "channel_reserve_satoshis": 0,
                    "dust_limit_satoshis": 546,
                    "funding_txid": "7669b0853a7be1563ba0b77c877c04a456795e2c26238a0a2ab6681988f2d1b6",
                    "htlc_minimum_msat": 0,
                    "max_accepted_htlcs": 483,
                    "max_htlc_value_in_flight_msat": 18446744073709551615,
                    "msatoshi_to_us": 80000000,
                    "msatoshi_total": 80000000,
                    "owner": "lightning_channeld",
                    "short_channel_id": "506368:1209:1",
                    "state": "CHANNELD_NORMAL",
                    "to_self_delay": 144
                }
            ],
            "connected": true,
            "id": "0371190acfb2e92bd1faa6ce4d12ff248798515a92ac903ac14d31b5172d9b2917",
            "log": [
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "got revoke 21: 0 changed",
                "got commitsig 22: feerate 20000, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "got revoke 22: 0 changed",
                "got commitsig 23: feerate 29115, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "got revoke 23: 0 changed",
                "got commitsig 24: feerate 29106, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "Got depth change 0->39 for 7669b0853a7be1563ba0b77c877c04a456795e2c26238a0a2ab6681988f2d1b6",
                "Funding tx 7669b0853a7be1563ba0b77c877c04a456795e2c26238a0a2ab6681988f2d1b6 depth 39 of 3",
                "Funding tx confirmed, but peer state CHANNELD_NORMAL lightning_channeld",
                "got revoke 24: 0 changed",
                "got commitsig 25: feerate 29108, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "got revoke 25: 0 changed",
                "got commitsig 26: feerate 29109, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "got revoke 26: 0 changed",
                "got commitsig 27: feerate 29114, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "got revoke 27: 0 changed",
                "got commitsig 28: feerate 29120, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "got revoke 28: 0 changed",
                "got commitsig 29: feerate 29146, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "got revoke 29: 0 changed",
                "got commitsig 30: feerate 29150, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "got revoke 30: 0 changed",
                "got commitsig 31: feerate 29155, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "got revoke 31: 0 changed",
                "got commitsig 32: feerate 29156, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "got revoke 32: 0 changed",
                "got commitsig 33: feerate 29160, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "got revoke 33: 0 changed",
                "got commitsig 34: feerate 27830, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "got revoke 34: 0 changed",
                "got commitsig 35: feerate 27827, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "got revoke 35: 0 changed",
                "got commitsig 36: feerate 27824, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "got revoke 36: 0 changed",
                "got commitsig 37: feerate 27827, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "got revoke 37: 0 changed",
                "got commitsig 38: feerate 27835, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "got revoke 38: 0 changed",
                "got commitsig 39: feerate 27828, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "got revoke 39: 0 changed",
                "got commitsig 40: feerate 26530, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "got revoke 40: 0 changed",
                "got commitsig 41: feerate 27832, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "got revoke 41: 0 changed",
                "got commitsig 42: feerate 27827, 0 added, 0 fulfilled, 0 failed, 0 changed",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in",
                "Peer transient failure in CHANNELD_NORMAL: Owning subdaemon lightning_channeld died (33792)",
                "Peer has reconnected, state CHANNELD_NORMAL",
                "Already have funding locked in"
            ],
            "netaddr": [
                "176.9.113.240:9735"
            ]
        }
    ]
}
`

**Comment by cdecker** [comment]
This was fixed by #1398 and #1404, should be good now.
